lib/src/yaml_node_wrapper.dart

```diff
@@ -32,9 +32,11 @@ class YamlMapWrapper extends MapBase
   @override
   Iterable get keys => _dartMap.keys;
 
-  YamlMapWrapper(Map dartMap, sourceUrl) : this._(dartMap, NullSpan(sourceUrl));
+  YamlMapWrapper(Map dartMap, sourceUrl,
+      {CollectionStyle style = CollectionStyle.ANY})
+      : this._(dartMap, NullSpan(sourceUrl), style: style);
 
-  YamlMapWrapper._(Map dartMap, SourceSpan span)
+  YamlMapWrapper._(Map dartMap, SourceSpan span, {this.style})
```

Comment:
This allows an `YamlMapWrapper` instance where `YamlMapWrapper.style == null`, this will actually happen in the `operator []` implementation which has `YamlMapWrapper._(value, span);`.

Perhaps the internal constructor `YamlMapWrapper._(...)` can just have `style` as a required parameter... or we give it a default value.

(I do no propose that we pass the style from parent to child in the `operator[]` implementation, instead just given `ANY`)

Reply:
I suppose we could consider a follow up PR where child elements that already implement `YamlNode` are not recursively wrapped in the `operator[]` implementation..

Such wrapping seems redundant, it does have side effects on the `SourceSpan`... So there is a few things to consider.

Reply:
Added default value to the internal constructor!